test/env/generate.go

```diff
@@ -14,7 +16,8 @@ import (
 )
 
 func GenerateLayer2WithSubnetNAD(nadName, namespace, role string) *nadv1.NetworkAttachmentDefinition {
-	networkName := "l2"
+	const randCharacters = 5
+	networkName := strings.Join([]string{"l2", role, rand.String(randCharacters)}, "-")
```

Comment:
I wonder if there are any chances of a collision here ...

I almost feel silly asking, but, shouldn't we cover against that ? 

Anyway ... I'm willing to try this out as is. We can adapt once we ever see a collision.

@RamLavi feel free to lift the hold I've set in another comment.

Reply:
IIUC rand.String(n) produces a set of n random characters. Avoiding vowels for some reason - each character can be one of 27. So chance of collison is 1/27^n  = 1/27^5 = 1/14,348,907 = 0.00000697%. We can live with that.